geometry.zsh

```diff
@@ -1,56 +1,96 @@
-# Geometry
-# Based on Avit and Pure
+# geometry - simple customizable prompt based on avit, pure, and mnml
+#
 # avit: https://github.com/robbyrussell/oh-my-zsh/blob/master/themes/avit.zsh-theme
 # pure: https://github.com/sindresorhus/pure
+# mnml: https://github.com/subnixr/minimal
+
 GEOMETRY_ROOT=${0:A:h}
-source "$GEOMETRY_ROOT/lib/async.zsh"
-source "$GEOMETRY_ROOT/lib/color.zsh"
-source "$GEOMETRY_ROOT/lib/grep.zsh"
-source "$GEOMETRY_ROOT/lib/plugin.zsh"
-source "$GEOMETRY_ROOT/lib/time.zsh"
-source "$GEOMETRY_ROOT/lib/title.zsh"
-
-# Flags
-PROMPT_GEOMETRY_SHOW_RPROMPT=${PROMPT_GEOMETRY_SHOW_RPROMPT:-true}
-PROMPT_GEOMETRY_RPROMPT_ASYNC=${PROMPT_GEOMETRY_RPROMPT_ASYNC:-true}
-PROMPT_GEOMETRY_ENABLE_PLUGINS=${PROMPT_GEOMETRY_ENABLE_PLUGINS:-true}
-PROMPT_GEOMETRY_PRIMARY_SUFFIX=${PROMPT_GEOMETRY_PRIMARY_SUFFIX:-" "}
-
-prompt_geometry_render() {
-  PROMPT="$(geometry_plugin_render primary)$PROMPT_GEOMETRY_PRIMARY_SUFFIX"
-
-  PROMPT2=" $GEOMETRY_SYMBOL_RPROMPT "
-
-  if $PROMPT_GEOMETRY_SHOW_RPROMPT; then
-    if $PROMPT_GEOMETRY_RPROMPT_ASYNC; then
-        # On render we reset rprompt until async process
-        # comes with newer git info
-        RPROMPT=""
-    else
-        setopt localoptions no_prompt_subst
-        RPROMPT="$(geometry_plugin_render secondary)"
-    fi
-  fi
+: ${GEOMETRY_SEPARATOR:=" "}
+
+(($+GEOMETRY_PROMPT)) || GEOMETRY_PROMPT=(geometry_status geometry_path)
+(($+GEOMETRY_RPROMPT)) || GEOMETRY_RPROMPT=(geometry_exec_time geometry_git geometry_hg)
```

Comment:
The defaults have changed significantly. I think that is okay, since it lets people know that 'things have changed', but I wonder if others feel differently and would rather it be the same out-of-box experience.

So let me know if we should change the defaults to be closer to the original


```
# move path back to left side
GEOMETRY_PROMPT=(geometry_status geometry_path)
GEOMETRY_RPROMPT=(geometry_exec_time geometry_git geometry_hg)
```

```
# add spaces around git separator again
GEOMETRY_GIT_SEPARATOR=" :: "
```

Reply:
@justmendes @MarioRicalde @desyncr